### PR TITLE
Read export from s3

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: data-migration/src/main/java/no/unit/nva/publication/migration/PublicationImporter.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        return BEGIN_ARRAY_DELIMITER + arrayElements + END_ARRAY_DELIMITER;
+    }
+
+    private static String makeConsecutiveJsonObjectsElementsOfJsonArray(String jsonObjects) {
+        return jsonObjects.replaceAll(CONSECUTIVE_JSON_OBJECTS, SUCCESSIVE_ELEMENTS_IN_ARRAY);
+    }
+
+    private static List<Publication> parseJson(String json) {
+        JsonNode root = attempt(() -> objectMapper.readTree(json)).orElseThrow();
+        if (root.isArray()) {
```
Comment (brinxmat) [inline data-migration/src/main/java/no/unit/nva/publication/migration/PublicationImporter.java:90]: Ternary leave single exit point

File: data-migration/src/test/java/no/unit/nva/publication/migration/PublicationImporterTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    private static final Path EXISTING_DATA_PATH = Path.of("AWSDynamoDB",
+        "01614600960660-a9739099", "data/");
+    private static final List<Publication> FIRST_PUBLICATIONS_FILE = samplePublications();
+    private static final List<Publication> SECOND_PUBLICATIONS_FILE = samplePublications();
+    private static final List<Publication> EXPECTED_PUBLICATIONS = constructExpectedPublications();
+    private PublicationImporter publicationImporter;
+
+    @BeforeEach
+    public void init() {
+        S3Driver remoteClient = remoteConnection();
```
Comment (brinxmat) [inline data-migration/src/test/java/no/unit/nva/publication/migration/PublicationImporterTest.java:47]: Unused?

File: data-migration/src/test/java/no/unit/nva/publication/migration/PublicationImporterTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+        private JsonNode addParentItem(JsonNode json) {
+            ObjectNode root = objectMapper.createObjectNode();
+            root.set(PublicationImporter.ION_ITEM, json);
+            return root;
+        }
+
+        private String convertJsonToIon(String json) {
+            var ionReader = IonReaderBuilder.standard().build(json);
+            var type = ionReader.next();
```
Comment (brinxmat) [inline data-migration/src/test/java/no/unit/nva/publication/migration/PublicationImporterTest.java:194]: Unused?

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by janbj — APPROVED** [review]
Looks good, but maybe add a comment about what input format this importer expects.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**